### PR TITLE
Static compilation fixes

### PR DESCRIPTION
**Pull request by ulrikstrid**

### CODE REVIEW

**Review by anmonteiro — DISMISSED** [review]
remove the ppx_deriving + uucp changes when rebasing pls

File: ocaml/default.nix [hunk 1]
```diff
@@ -930,6 +934,7 @@ with oself;
       ppx_derivers
       result
     ];
+    buildInputs = [ ];
```
Comment (anmonteiro) [inline ocaml/default.nix:937]: This is now duplicatewd. check line 931
```suggestion

```

File: ocaml/default.nix [hunk 2]
```diff
@@ -1154,6 +1163,10 @@ with oself;
     };
   });
 
+  uucp = osuper.uucp.overrideAttrs (o: {
+    buildInputs = o.buildInputs ++ [ uucd ];
+  });
+
```
Comment (anmonteiro) [inline ocaml/default.nix]: this is fixed
```suggestion

```

File: ocaml/default.nix [hunk 3]
```diff
@@ -253,7 +254,7 @@ with oself;
     '';
   });
 
-  ctypes-0_17 = osuper.ctypes.overrideAttrs (_: {
+  ctypes-0_17 = osuper.ctypes.overrideAttrs (o: {
```
Comment (anmonteiro) [inline ocaml/default.nix:257]: if you make this inherit from `oself.ctypes` then you can inherit the fix from the other one

File: ocaml/default.nix [hunk 4]
```diff
@@ -259,7 +261,7 @@ with oself;
     '';
   });
 
-  ctypes-0_17 = osuper.ctypes.overrideAttrs (_: {
+  ctypes-0_17 = oself.ctypes.overrideAttrs (o: {
```
Comment (anmonteiro) [inline ocaml/default.nix:264]:
```suggestion
  ctypes-0_17 = oself.ctypes.overrideAttrs (_: {
```

File: ocaml/default.nix [hunk 5]
```diff
@@ -405,6 +409,10 @@ with oself;
     };
   });
 
+  ff-pbt = osuper.ff-pbt.overrideAttrs (o: {
+    propagatedBuildInputs = o.propagatedBuildInputs ++ [ alcotest ];
```
Comment (anmonteiro) [inline ocaml/default.nix:413]: don't propagate a test dependency I guess? maybe add to buildInputs

Comment (ulrikstrid) [reply ocaml/default.nix:413]: Yes :+1:

**Review by anmonteiro — APPROVED** [review]

File: ocaml/default.nix [hunk 6]
```diff
@@ -405,6 +409,10 @@ with oself;
     };
   });
 
+  ff-pbt = osuper.ff-pbt.overrideAttrs (o: {
+    propagatedBuildInputs = o.buildInputs ++ [ alcotest ];
```
Comment (anmonteiro) [inline ocaml/default.nix:413]:
```suggestion
    propagatedBuildInputs = o.propagatedBuildInputs ++ [ alcotest ];
```

**Pull request merged**